stlab/concurrency/future.hpp

```diff
@@ -1705,8 +1685,8 @@ auto when_any(E executor, F&& f, std::pair<I, I> range) {
 /**************************************************************************************************/
 
 template <typename E, typename F, typename... Args>
-auto async(E executor, F&& f, Args&&... args)
-    -> future<detail::result_t<std::decay_t<F>, std::decay_t<Args>...>> {
+auto async(E executor, F&& f, Args&&... args) -> future<
```

Comment:
Let's discuss this as a breaking change on the (public) stlab slack - (let me know if you need an invite). Personally, I think I'm fine with it as I doubt there are any (or many) cases this would break and likely the break is minor but we might consider alternate names or using namespace versioning.

Reply:
Yes, this need to be discussed. I think need an invite, I can't sign in into stlab.slack.com.